joerd/check.py

```diff
@@ -19,6 +20,21 @@ def is_zip(tmp):
     return False
 
 
+def is_tar_gz(tmp):
+    """
+    Returns True if the NamedTemporaryFile given as the argument appears to be
+    a GZip-encoded TAR file.
+    """
+
+    try:
+        tar = tarfile.open(tmp.name, mode='r:gz', errorlevel=2)
+        names = [info.name for info in tar]
```

Comment:
I want to check if the tarball is valid, so I'm _hoping_ (given an absence of an explicit method to check the checksums) that enumerating all the files will throw errors if the file is corrupt. I have a sneaking suspicion that it won't and it'll only find it when the file is extracted... perhaps I should do that instead.